Firestore/core/src/firebase/firestore/nanopb/byte_string.h

```diff
@@ -39,6 +44,11 @@ class ByteString : public util::Comparable<ByteString> {
  public:
```

Comment:
I wasn't sure I agreed when I started to try to follow this advice, but in the end I like it more.

Unfortunately, the class was so large because it was trying to make a bunch of disparate use cases fit together without changing the sources that much. Fixing this required fixing `ByteString` users to be more uniform.

For example, `ByteString` had constructors/conversions to deal with `std::vector` because that was how `ByteStringWriter` accumulated its contents. However, making `ByteStringWriter` directly accumulate in a `pb_bytes_array_t*` eliminates this need (and also saves a copy).

I've also made the conversions free functions since they can depend on the public interface.